surface/include/pcl/surface/advancing_front_utils.h

```diff
@@ -0,0 +1,310 @@
+    bool
+    alignNormal (Eigen::Vector3f &pn, const Eigen::Vector3f &av)
+    {
+      double dot = pn.dot (av);
+      if (dot < 0.0)
+      {
+        pn *= -1.0;
+        return true;
+      }
+
+      return false;
+    }
+
+    /** \brief Align pn's normal with av so they point in the same direction */
+    bool
+    alignNormal (AfrontVertexPointType &pn, const AfrontVertexPointType &av)
+    {
+      Eigen::Vector3f normal = pn.getNormalVector3fMap ();
+      if (alignNormal (normal, av.getNormalVector3fMap ()))
+      {
+        pn.normal_x = normal[0];
+        pn.normal_y = normal[1];
+        pn.normal_z = normal[2];
+        return true;
+      }
+
+      return false;
+    }
+
+    /** \brief Check if two normals are within a tolerance */
+    bool
+    checkNormal (const Eigen::Vector3f n1, const Eigen::Vector3f n2, const double &tol)
+    {
+      double dot = n1.dot (n2);
+      if (dot < 0.0)
+        return false;
+
+      double denom = n1.norm () * n2.norm ();
+      double angle = acos (dot / denom);
+      if (angle > tol)
+        return false;
+
+      return true;
+    }
+
+    /** \brief Convert Eigen Vector3f to PCL PointXYZ */
+    pcl::PointXYZ
+    convertEigenToPCL (const Eigen::Vector3f &p)
+    {
+      return pcl::PointXYZ (p (0), p (1), p (2));
+    }
+
+    /** \brief Convert Eigen Vector3f to PCL PointXYZ */
+    pcl::PointXYZ
+    convertEigenToPCL (const Eigen::Vector3d &p)
+    {
+      return pcl::PointXYZ (p (0), p (1), p (2));
+    }
+
+    AfrontVertexPointType
+    convertPointNormalToAfrontPointType (const pcl::PointNormal &p)
+    {
+      AfrontVertexPointType result;
+      result.x = p.x;
+      result.y = p.y;
+      result.z = p.z;
+      result.data[3] = 1.0f;
+      result.normal_x = p.normal_x;
+      result.normal_y = p.normal_y;
+      result.normal_z = p.normal_z;
+      result.data_n[3] = 0.0f;
+      result.curvature = p.curvature;
+      result.max_step = 0.0f;
+      result.max_step_search_radius = 0.0f;
+      return result;
+    }
+
+    AfrontGuidanceFieldPointType
+    convertAfrontPointTypeToAfrontGuidanceFieldPointType (const AfrontVertexPointType &p, const double rho)
+    {
+      pcl::afront::AfrontGuidanceFieldPointType result;
+      result.x = p.x;
+      result.y = p.y;
+      result.z = p.z;
+      result.data[3] = 1.0f;
+      result.normal_x = p.normal_x;
+      result.normal_y = p.normal_y;
+      result.normal_z = p.normal_z;
+      result.data_n[3] = 0.0f;
+      result.curvature = p.curvature;
+      result.ideal_edge_length = 2.0 * std::sin (rho / 2.0) / result.curvature;
+      return result;
+    }
+
+    /**
+      * \brief Get the mid point of a half edge given it's verticies
+      * \param[in] p1 Vertex of half edge
+      * \param[in] p2 Vectex of half edge
+      * \return The mid point of the half edge
+      */
+    Eigen::Vector3f
+    getMidPoint (const Eigen::Vector3f &p1, const Eigen::Vector3f &p2)
```

Comment:
This is really trivial and is only used in two places, I'd simply expand this at the points of usage.